feat(linter): support `ignorePatterns` for nested configs

Sysix opened the pull request:
Now every `ignorePatterns` configuration in nested configuration are respected.
Before, only `ignorePatterns`, which are located in the base-configuration of the project, would be respected.

Closes #11067
Closes #12857
Closes #11969

Sysix commented on apps/oxlint/src/snapshots/fixtures__config_ignore_patterns__with_oxlintrc_-c .__test__eslintrc.json --ignore-pattern _.ts .@oxlint.snap:
```diff
@@ -5,13 +5,7 @@ source: apps/oxlint/src/tester.rs
-  ! ]8;;https://oxc.rs/docs/guide/usage/linter/rules/unicorn/no-empty-file.html\eslint-plugin-unicorn(no-empty-file)]8;;\: Empty files are not allowed.
-   ,-[main.js:1:1]
-   `----
-  help: Delete this file or add some code to it.
```
In `eslintrc.json` is `*.js` ignored. This looks like a fix for me.
When someone uses a configuration file in a directory with `-c`, the `ignorePatterns` should not adjust to its directory (`/test/`)

Sysix commented:
@camc314 /ping

camc314 commented on apps/oxlint/src/snapshots/_-c fixtures__config_ignore_patterns__ignore_extension__eslintrc.json fixtures__config_ignore_patterns__ignore_extension__main.js@oxlint.snap:
```diff
@@ -5,7 +5,8 @@ source: apps/oxlint/src/tester.rs
+Finished in <variable>ms on 1 file with 88 rules using 1 threads.
 ----------
-CLI result: LintNoFilesFound
+CLI result: LintSucceeded
```
It feels like this should still say `on 0 files`... since it didn't actually lint anything (`fixtures/config_ignore_patterns/ignore_extension/main.js` should've been in the ignore pattern here > https://github.com/oxc-project/oxc/blob/32f860d238ca97dc8691b5fcec680d99ad9aab42/apps/oxlint/fixtures/config_ignore_patterns/ignore_extension/eslintrc.json#L2-L3)

camc314 commented on crates/oxc_linter/src/service/runtime.rs:
```diff
@@ -274,6 +274,17 @@ impl Runtime {
     ) {
         if self.resolver.is_none() {
             self.paths.par_iter().for_each(|path| {
+                if self
```
I think we may still want to process it here, so that we can add the modules info to the module graph (regardless of whether it's actaully going to be linted.)

camc314 replied:
🤦 haha i should've read L 275

camc314 replied:
> Will fix it in line 380 by moving it before the module_to_lint.push

yes this change seems good to me, thanks

Sysix replied:
With your request of the real linted file numbers. I am moving the scope to the tools (oxlint / language server).
The server does not need to know the numbers of files.

Sysix commented:
@camc314 Sorry to bother, but I am getting a bit frustrated resolving the merge conflicts >.<

camc314 reviewed (approved):
sorry for the wait! thanks for fixing this!

Sysix committed:
feat(linter): support `ignorePatterns` for nested configs (#12210)

Now every `ignorePatterns` configuration in nested configuration are respected.
Before, only `ignorePatterns`, which are located in the base-configuration of the project, would be respected.

Closes #11067
Closes #12857
Closes #11969
